src/player.cpp

```diff
@@ -536,6 +536,20 @@ PlayerErrors Player::play(
         return bufferStreamCanBePlayedOnlyOnce;
     }
 
+    // Check if by playing this sound will exceed the maximum number of voice count. If true, then
+    // check if [soudHash] has other instances playing. If true remove the first and play the new one.
+    // If there are no other instances playing, this sound cannot be played and return an error.
+    // Issue https://github.com/alnitak/flutter_soloud/issues/204
+    if (getActiveVoiceCount() >= getMaxActiveVoiceCount())
+    {
+        if (sound->handle.size() > 0)
+        {
+            stop(sound->handle[0].handle);
```

Comment:
Are we absolutely certain that this is safe? Is it possible the sound is already stopped? Or there's a null somewhere in here?

Also, are we sure that `handle[0]` is the oldest handle?

Reply:
Also, are we sure that stopping here will immediately clear the voice for the sound to be played later on line 554?

Reply:
If any of this isn't 100% crystal clear, or if there's a possibility this will be a maintainability issue, I'd strongly suggest we just drop the new sound and return the error, not trying to play it. Again, we're talking about an edge case. We don't need to go out of our way to make this work, and I worry this will be a bug magnet.